instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/context/ContextPropagationDebug.java

```diff
@@ -50,27 +50,32 @@ public static void debugContextLeakIfEnabled() {
       if (currentSpan != null) {
         log.error("It contains this span: {}", currentSpan);
       }
-      List<StackTraceElement[]> locations = ContextPropagationDebug.getLocations(current);
-      if (locations != null) {
-        StringBuilder sb = new StringBuilder();
-        Iterator<StackTraceElement[]> i = locations.iterator();
-        while (i.hasNext()) {
-          for (StackTraceElement ste : i.next()) {
-            sb.append("\n");
-            sb.append(ste);
-          }
-          if (i.hasNext()) {
-            sb.append("\nwhich was propagated from:");
-          }
-        }
-        log.error("a context leak was detected. it was propagated from:{}", sb);
-      }
+
+      debugContextPropagation(current);
 
       if (FAIL_ON_CONTEXT_LEAK) {
         throw new IllegalStateException("Context leak detected");
       }
     }
   }
 
+  public static void debugContextPropagation(Context context) {
```

Comment:
```suggestion
  private static void debugContextPropagation(Context context) {
```

Don't see it used

Reply:
I used it during the debugging of this PR. That was the reason for extraction as a separate method.

Reply:
Well if we keep it public we need `// Visible for testing` maybe to make that clear :)

Reply:
I mean, I can make it private. Then whenever a developer wants to use it during debugging, they just make it public again :)

Reply:
Ah - yeah that's fine, it's why my suggestion does that :P